src/tasks/containers/TasksPage.test.tsx

```diff
@@ -180,4 +239,103 @@ describe('Tasks.Containers.TasksPage', () => {
       )
     })
   })
+
+  describe('manage tasks', () => {
+    it('deletes a task', async () => {
+      const taskCard = (await screen.findAllByTestId('task-card'))[0]
+
+      expect(
+        taskCard.querySelector("[data-testid='task-card--name']").textContent
+      ).toContain('Dead Beetle')
+
+      const deleteButton = taskCard.querySelector(
+        "[data-testid='context-delete-task']"
+      )
+
+      const taskID = taskCard
+        .querySelector("[class='copy-resource-id']")
+        .textContent.split(':')[1]
+        .split('C')[0]
+        .trim()
+
+      expect(ui.store.getState().resources.tasks.byID[taskID]).toBeTruthy()
+      expect(ui.store.getState().resources.tasks.allIDs).toContain(taskID)
+
+      fireEvent.click(deleteButton)
+
+      await waitFor(() => expect(deleteTask).toBeCalled())
+
+      expect(mocked(deleteTask).mock.calls[0][0]['taskID']).toEqual(taskID)
+      expect(ui.store.getState().resources.tasks.byID[taskID]).toBeFalsy()
+      expect(ui.store.getState().resources.tasks.allIDs).not.toContain(taskID)
+    })
+
+    it('clones a task', async () => {
+      expect(ui.store.getState().resources.tasks.allIDs.length).toEqual(
+        localTasks.length
+      )
+
+      const taskCard = (await screen.findAllByTestId('task-card'))[0]
+      const cloneButton = taskCard.querySelector(
+        '[data-testid=context-menu] + div [data-testid=context-menu-item]'
+      )
+      const name = taskCard.querySelector("[data-testid='task-card--name']")
+        .textContent
+      expect(name).toContain(InactiveTask.name)
+
+      fireEvent.click(cloneButton)
+
+      await waitFor(() => expect(postTask).toBeCalled())
+
+      expect(mocked(getTask).mock.calls[0][0].taskID).toEqual(InactiveTask.id)
+
+      expect(mocked(parse).mock.calls[0][0]).toEqual(sampleScript)
+
+      expect(mocked(postTask).mock.calls[0][0].data.id).toEqual(InactiveTask.id)
+      expect(mocked(postTask).mock.calls[0][0].data.name).toEqual(
+        InactiveTask.name
+      )
+      expect(mocked(postTask).mock.calls[0][0].data.status).toEqual(
+        InactiveTask.status
+      )
+
+      expect(ui.store.getState().resources.tasks.allIDs.length).toEqual(
+        localTasks.length + 1
+      )
+      expect(ui.store.getState().resources.tasks.allIDs).toContain(
+        replacementID
+      )
+      expect(
+        ui.store.getState().resources.tasks.byID[replacementID].name
+      ).toEqual(replacementName)
+    })
+
+    /*
+    NB was working up until most recent rebase 0bfc0bb370af 21.6.16
+    Now throws error on fire event
+
+    [Util] handleError::  Cannot assign to read only property 'status' of object '#<Object>'
+     :-/
+    */
+    it.skip('activates a task', async () => {
```

Comment:
@karel-rehor I think this error started after we updated the `immer` library recently. It became stricter about modifying component props directly. I was able to fix the issue in [this commit](https://github.com/influxdata/ui/commit/200af228869f7759d6ef5452681237edb8680328). I don't have permission to modify this fork so I'll leave it to you to bring those changes into this PR